core/src/main/java/com/linecorp/armeria/common/multipart/MimeParser.java

```diff
@@ -230,7 +238,15 @@ void parse() {
                         state = State.BODY;
                         startOfLine = true;
 
-                        bodyPartPublisher = new DefaultStreamMessage<>();
+                        bodyPartPublisher = new DefaultStreamMessage<HttpData>() {
+                            @Override
+                            protected void onRequest(long n) {
+                                // Should we limit this to 1? Because downstream can request aggressively
+                                // and ignore the demand number of BodyPart. Or how can we control
+                                // this stop when we meet END_PART.
+                                onBodyPartRequestHttpData.accept(n);
```

Comment:
Maybe I need to store the `n` and request one by one at `state == BODY` case below.